src/BaselineUtils.cc

```diff
@@ -33,10 +33,100 @@ const int deblend::BaselineUtils<ImagePixelT, MaskPixelT, VariancePixelT>::STRAY
 template <typename ImagePixelT, typename MaskPixelT, typename VariancePixelT>
 const int deblend::BaselineUtils<ImagePixelT, MaskPixelT, VariancePixelT>::STRAYFLUX_TRIM;
 
-static bool span_ptr_compare(PTR(det::Span) sp1, PTR(det::Span) sp2) {
-    return (*sp1 < *sp2);
+static bool span_compare(geom::Span const & sp1,
+                         geom::Span const & sp2) {
+    return (sp1 < sp2);
 }
 
+namespace {
+    void nearestFootprint(std::vector<PTR(det::Footprint)> const& foots,
+                          image::Image<std::uint16_t>::Ptr argmin,
+                          image::Image<std::uint16_t>::Ptr dist)
+    {
+        /*
+         * insert the footprints into an image, set all the pixels to the
+         * Manhattan distance from the nearest set pixel.
+         */
+        typedef std::uint16_t dtype;
+        typedef std::uint16_t itype;
+
+        const itype nil = 0xffff;
+
+        *argmin = 0;
+        *dist = 0;
+
+        {
+        int i = 0;
+        for (std::shared_ptr<det::Footprint> const & foot : foots) {
+            foot->getSpans()->setImage(*argmin, static_cast<itype>(i));
+            foot->getSpans()->setImage(*dist, static_cast<dtype>(1));
+            ++i;
+        }
+        }
```

Comment:
I don't know in what sense you mean better. From a code efficiency standpoint, it will almost certainly compile out to the same thing. If you mean from a readability standpoint, maybe? If it were a standard for loop it would just move the lines with i in them into the line that declares the loop, and you would then either need to have an iterator object to deference, or use the [] syntax with the vector all over the place. I don't think there is a clear winner, and I preferred how this looked, rather than having a complicated for line. It would be nice if c++ supported enumerations in the range based for loop. Unfortunately we will need to wait until c++ 17 for that I think.